### PR TITLE
test(numeral-date): refactor for cypress tests

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

Refactor Cypress tests for the `Numeral-Date` component to use new cypress-component-react framework for testing.
Add `numeral-date.test.js`. 

### Current behaviour

Current process is the regular Cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `numeral-date.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

**Review by ZhuoyuJin — APPROVED** [review]

File: src/components/numeral-date/numeral-date.test.js [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+
+      numeralDateInputByPosition(0)
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+    });
+
+    it("should render NumeralDate with readOnly prop", () => {
+      CypressMountWithProviders(<NumeralDateComponent readOnly />);
+
+      numeralDateInputByPosition(0).and("have.attr", "readOnly");
```
Comment (stephenogorman) [inline src/components/numeral-date/numeral-date.test.js:204]: numeralDateInputByPosition(0).should("have.attr", "readOnly");

File: src/components/numeral-date/numeral-date.test.js [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+          />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .should("have.css", "width", labelRatio);
+
+        getDataElementByValue("input")
+          .parent()
+          .should("have.css", "width", "58px");
```
Comment (stephenogorman) [inline src/components/numeral-date/numeral-date.test.js:175]: Is inputWidth a prop that can be changed?

File: src/components/numeral-date/numeral-date.test.js [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+    it("should render NumeralDate with defaultValue prop", () => {
+      CypressMountWithProviders(
+        <NumeralDate defaultValue={{ dd: "12", mm: "12", yyyy: "2022" }} />
+      );
+
+      numeralDateInputByPosition(0).should("have.attr", "value", "12");
+      numeralDateInputByPosition(1).should("have.attr", "value", "12");
+      numeralDateInputByPosition(2).should("have.attr", "value", "2022");
+    });
+
```
Comment (stephenogorman) [inline src/components/numeral-date/numeral-date.test.js:265]: Also need a test for the value prop.

File: src/components/numeral-date/numeral-date.test.js (hunk 2 above)
Comment (DlgSHi) [reply src/components/numeral-date/numeral-date.test.js]: deleted

File: src/components/numeral-date/numeral-date.test.js (hunk 1 above)
Comment (DlgSHi) [reply src/components/numeral-date/numeral-date.test.js]: fixed

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**